Switch to using cim go images

freeformzSFDC opened the pull request:
https://hub.docker.com/r/cimg/go
https://github.com/CircleCI-Public/cimg-go
https://circleci.com/blog/announcing-our-next-generation-convenience-images-smaller-faster-more-deterministic/

Commit:
Switch to using cim go images

https://hub.docker.com/r/cimg/go
https://github.com/CircleCI-Public/cimg-go
https://circleci.com/blog/announcing-our-next-generation-convenience-images-smaller-faster-more-deterministic/

danp commented on src/golang/executors/golang-with-dbs.yml:
```diff
@@ -1,13 +1,13 @@
+    description: tag of the circleci/golang docker image to use (Defaults to '1.14')
     type: string
-    default: latest
+    default: "1.14"
```
Do wonder if this should be required so it's not forgotten as part of version bumps.

freeformzSFDC replied:
probably.

Commit:
golang/executors: No default go version

AFAICT circle ci isn't publishing a 'latest' tag of the cim/go images
and generally speaking we always want to specify the go version anyway.
